gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceLauncher.java

```diff
@@ -29,4 +33,36 @@
   /** The instance name (for debugging/logging purposes) */
   String getInstanceName();
 
+  // Standard configuration for Gobblin instances
+  @Getter
+  public static class ConfigAccessor {
+    static final String RESOURCE_NAME =
+        GobblinInstanceLauncher.class.getPackage().getName().replaceAll("[.]", "/")
+        + GobblinInstanceLauncher.class.getSimpleName()
+        + ".conf";
+    /** The namespace for all config options */
+    static final String CONFIG_PREFIX = "gobblin.instance";
+    /** The time to wait for gobblin instance components to start in milliseconds. */
+    static final String START_TIMEOUT_MS = "startTimeoutMs";
+    /** The time wait for Gobblin components to shutdown in milliseconds. */
+    static final String SHUTDOWN_TIMEOUT_MS = "shutdownTimeoutMs";
+
+    private final long startTimeoutMs;
+    private final long shutdownTimeoutMs;
+
+    /** Config accessor from a no namespaced typesafe config. */
+    public ConfigAccessor(Config cfg) {
+      Config effectiveCfg = cfg.withFallback(getDefaultConfig());
+      this.startTimeoutMs = effectiveCfg.getLong(START_TIMEOUT_MS);
+      this.shutdownTimeoutMs = effectiveCfg.getLong(SHUTDOWN_TIMEOUT_MS);
+    }
+
+    public static Config getDefaultConfig() {
+      return ConfigFactory.parseResources(RESOURCE_NAME).withFallback(ConfigFactory.load());
```

Comment:
you can use `ConfigFactory.parseResources(GobblinInstanceLauncher.class, GobblinInstanceLauncher.class.getSimpleName())` instead